framework/include/samplers/Sampler.h

```diff
@@ -190,18 +228,45 @@ class Sampler : public MooseObject, public SetupInterface, public DistributionIn
   /// Initial random number seed
   const unsigned int & _seed;
 
-  /// Data offsets for computing location based on global row index
-  std::vector<unsigned int> _offsets;
-
-  /// Total number of rows
-  dof_id_type _total_rows;
-
-  /// Number of global rows for this processor
-  dof_id_type _local_rows;
+  /// Number of rows for this processor
+  dof_id_type _n_local_rows;
 
   /// Global row index for start of data for this processor
   dof_id_type _local_row_begin;
 
   /// Global row index for end of data for this processor
   dof_id_type _local_row_end;
+
+  /// Total number of rows in the sample matrix
+  dof_id_type _n_rows;
```

Comment:
This is generally not the right type for length type holders. Our desktop build still defaults to 32-bit while container types can hold 64-bit counters.